olp-cpp-sdk-core/src/cache/DiskCache.cpp

```diff
@@ -120,6 +120,12 @@ client::ApiError GetApiError(const leveldb::Status& status) {
   return client::ApiError(code, status.ToString());
 }
 
+int GetFilesNumberAtLevel0(leveldb::DB& db) {
```

Comment:
Why not `CheckCompactionFinished` or `ContinueCompaction`, as we are not really interested in the number of L0 or L1 files but if we should keep up the compaction. Also, you can move the log here. Later if we discover that it is not enough and also L1 files could be a problem we, can easily extend this method with another check instead of adding another method with another check below.